### ISSUE TITLE
version from git tag is now preferred over package.json ? breaks custom git tag formats

### ISSUE THREAD

**Issue by mark-buer** [webpro/release-it#116]
`release-it` now does not run unless the git tag is "well formed" (as-in, can be parsed as a semver).

It seems that the logic surrounding the version determination has altered in a recent refactoring.

In changeset https://github.com/webpro/release-it/commit/6d46d9046cf6fd0944140066fc1c64584674de1a

the logic changed for determining the version from something a little like:
```
function parseVersion() {

  const options = config.options;
  const version = util.isValidVersion(options.increment) ? options.increment : options.npm.version;

  if(!version) {

    return git.getLatestTag().then(tag => {
      if(tag) {
        const nextVersion = util.increment(tag, options.increment, options.prereleaseId);
```

to something a little like this:
```
export async function parseVersion(options) {
  const { increment, preReleaseId } = options;

  const latestVersion = (await getLatestTag()) || options.npm.version;

  const isValidInc = isValidVersion(increment) && semver.gt(latestVersion, increment);
  const isValidType = _.includes(releaseTypes, increment);
```

In summary, it was previously the case that the version from `package.json` was preferred, but now the version from the git tag is preferred.

This breaks anyone using a custom git tagging scheme.

**Comment by webpro** [comment]
Hi @mark-buer, thanks for bringing this up.

This project follows semver, so the bump to 3.0.0 was indeed be a breaking change in some cases. Unfortunately this was also true for your project(s).

Apart from whether the latest version should be derived from the Git tag or npm package version, I didn't take custom tagging schemes into account during the rewrite, as I didn't realize this is a thing at the time. Do you have some references or examples for this?

In any case, I've just released [v3.0.2](https://github.com/webpro/release-it/releases/tag/3.0.2) which validates the latest git tag before falling back to the npm package's version. Please let me know if this works for you.

**Comment by webpro** [comment]
Closing due to inactivity. I think the fix was adequate.
